### DISCUSSION TITLE
Saving users and groups

### DISCUSSION THREAD

**Discussion by eyearian** [milvus-io/milvus#30605] [Q&A and General discussion]
Hi - does the backup utility backup users and groups?  Is there a way save these users and groups if not?

Ideally I would like to way declaratively create/store these users and groups to be saved in source control if the backup utility doesn't support this.  This would help with easily being able to recreate a cluster and its users without having to manually create each user and group again.

**Comment by yhmo** [comment]
Milvus backup tool cannot backup users/roles, it only backup/restore collections.
Users/roles are stored in Etcd service. So far we have no plan to support users/roles backup.
